helper/helper_test.go

```diff
@@ -98,4 +294,234 @@ var _ = Describe("helper tests", func() {
+			testDir := "" // Use local directory for the TOC file instead of default
+			*tocFile = fmt.Sprintf("%stest_toc.yaml", testDir)
+			writeTestTOC(*tocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			*dataFile = "test_data.dat"
+			// Call the function under test
+			err := doRestoreAgentInternal(mockHelper)
+
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("successfully restores using multiple data files when inputs are valid and no errors occur", func() {
+			// Call the function under test
+			oidBatch := []oidWithBatch{{oid: 1, batch: 1}}
+			steps := []helperTestStep{
+				{"mock_1_1", true, 1, false, "restores using multiple data files"},
+			}
+
+			mockHelper := newHelperTest(oidBatch, steps)
+
+			err := doRestoreAgentInternal(mockHelper)
+
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("skips batches with corresponding skip file in doRestoreAgent", func() {
+			// Test Scenario 1. Simulate 1 pass for the doRestoreAgent() function with the specified oids, batches and expected calls
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Can open pipe for table 100, check_skip_file shall not be called"},
+				{"mock_200_0", true, -1, false, "Can open pipe for table 200, check_skip_file shall not be called"},
+				{"mock_200_1", false, 200, true, "Can not open pipe for table 200, check_skip_file shall called, skip file exists"},
+				{"mock_200_2", true, -1, false, "Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).To(BeNil())
+		})
+		It("skips batches if skip file is discovered with resize restore", func() {
+			*isResizeRestore = true
+			*origSize = 3
+			*destSize = 5
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Can open pipe for table 100, check_skip_file shall not be called"},
+				{"mock_200_0", true, -1, false, "Can open pipe for table 200, check_skip_file shall not be called"},
+				{"mock_200_1", false, 200, true, "Can not open pipe for table 200, check_skip_file shall called, skip file exists"},
+				{"mock_200_2", true, -1, false, "Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper)
+			Expect(err).To(BeNil())
+		})
+		It("skips batches if skip file is discovered with single datafile", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
```

Comment:
What is difference from test "successfully restores using a single data file when inputs are valid and no errors occur" above?

Reply:
Sorry, don't understand the question. The scenario is totally different.

Reply:
I didn't ask about the whole scenario, but only about working with the TOC-file.

Reply:
Then I don't understand your question.

Reply:
test "successfully restores using a single data file when inputs are valid and no errors occur" creates TOC-file and this test creates TOC-file but by other way, why?

Reply:
This one has a few batches and a skip file in the scenario, the other one has one batch and no skip file. Both tests are not for a TOC but using it.

Reply:
> Both tests are not for a TOC but using it.

Why can't the same TOC-files be used in both tests?

Reply:
May be they can, there was no such an idea/requirement. Should they?

Reply:
I'm asking if it's possible to use commands like these
```go
			*tocFile = testTocFile
			writeTestTOC(testTocFile)
```
instead of these
```go
			testDir := ""
			*tocFile = fmt.Sprintf("%stest_toc.yaml", testDir)
			writeTestTOC(*tocFile)
```
in the "successfully restores using a single data file when inputs are valid and no errors occur" test?

Reply:
okay